src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/math/SynchronizedUnivariateSolver.java

```diff
@@ -0,0 +1,385 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.math;
```

Comment:
Oh, this is a tricky one. Please read the class description carefully and let's discuss more if you're still not convinced of its (important) use case :-) here's the problem in a nutshell: you have some legacy iterative code that is implemented like:
```
void run() {
    while (true) {
        \\ do stuff
        \\ break if a condition is met
    }
}
```
Inside this run-forever block lies an external function call. Now, you want to run many of these routines in parallel, and you want to minimize the number of function calls. For example, if each function call is a Spark map-reduce, it comes with a large overhead; however, if you bundle several map-reduce operations into one "combined" call, you can pay the overhead price only once. This is what `SynchronizedUnivariateSolver` achieves, and it is quite non-trivial. The idea is to make "lazy" function calls, keep track of the function calls of each routine by locking its thread, dispatch a "bundled" call every once in a while, distribute the queries back to the routines, and unlock their thread; repeat until they all converge.

ps> the code has been significantly simplified in some of the latest commits.